target/linux/ipq807x/base-files/lib/preinit/10_fix_eth_mac.sh

```diff
@@ -0,0 +1,16 @@
+. /lib/functions/system.sh
+
+preinit_set_mac_address() {
+	case $(board_name) in
+	zyxel,nbg7815)
+		ethaddr=$(mtd_get_mac_ascii "0:appsblenv" ethaddr)
+		ip link set dev lan2 address $ethaddr
+		ip link set dev lan3 address $ethaddr
+		ip link set dev lan4 address $ethaddr
+		ip link set dev 10g address $ethaddr
+		ip link set dev wan address $(macaddr_add $ethaddr 1)
```

Comment:
why this is needed? mac are not assigned by uboot and copied to the local-mac-address binding?

Reply:
Even if not set, it should be possible to use NVMEM-s U-boot driver to read the cell directly in DTS

Reply:
> Even if not set, it should be possible to use NVMEM-s U-boot driver to read the cell directly in DTS

I tried last weekend and it was not working. I do not know if it was the dynamically declared u-boot env partition or missing nvmem support in the network driver.

Reply:
> why this is needed? mac are not assigned by uboot and copied to the local-mac-address binding?

There only one mac address in u-boot env. That's why it only works for eth0 / lan1.

Reply:
afaik the uboot nvmem driver still require some love to correctly work... for now i would use the uboot way that is probably supported

Reply:
> > Even if not set, it should be possible to use NVMEM-s U-boot driver to read the cell directly in DTS
> 
> I tried last weekend and it was not working. I do not know if it was the dynamically declared u-boot env partition or missing nvmem support in the network driver.

Just that that you fixed it by specifying the partitions manually. Will do that.

Reply:
Cloud

> Even if not set, it should be possible to use NVMEM-s U-boot driver to read the cell directly in DTS

Could you take a look at the dts? I've changed it, added the kernel symbol for nvmem uboot-env. But always random macs...

Reply:
That's the error in dmesg:
OF: /soc/dp1: could not get #nvmem-cell-cells for /soc/spi@78b5000/flash@0/partitions/partition@600000/ethaddr

Reply:
in theory should be just a wanr to ignore. Does the mac gets correctly assigned?

Reply:
@ddimension can you give output of fw_printenv ?

Reply:
Please try to use ```mac-address-ascii``` instead of ```mac-address``` in ```nvmem-cell-names```. Currently, Linux Kernel doesn't support ascii-based MAC address via NVMEM and the downstream implementation of ```of_get_mac_address``` is necessary.

example: https://github.com/openwrt/openwrt/pull/11228/files?diff=unified&w=0#diff-1fd579c71484b41770913dd23ce55bbb7f73dc034a8636d48239154d3c770ccbR209

Reply:
That NVMEM warning is false and there is a patch pending to fix it:
https://lore.kernel.org/lkml/DU0PR04MB941732E9BB61E51ADD66F82088FC9@DU0PR04MB9417.eurprd04.prod.outlook.com/T/

Also, Rafal has been working on adding ASCII support to NVMEM

Reply:
Yepp, it works :-)